lib/sinicum/imaging/imaging.rb

```diff
@@ -90,8 +94,17 @@ def perform_conversion
             convert(in_file.path, out_file.path)
             FileUtils.mv(out_file.path, file_rendered)
             FileUtils.chmod(0644, file_rendered)
+            out_index = out_file.path.rindex("-out-")
```

Comment:
Still too complicated. I think, that `perform_conversion` should stay exactly the same like before.

You will need to change the `file_out` method, to handle the sourceset option. Besides this, I would use a `srcset_if_needed` method like this:

```
def srcset_option_if_needed
  @srcset_option.nil? ? "" : "_#{@srcset_option}"
end
```